bibtex-actions.el

```diff
@@ -607,21 +584,16 @@ If no citation key is found, target entries can be chosen
 interactively when `bibtex-actions-at-point-fallback' is non-nil.
 With prefix ARG, rebuild the cache before offering candidates."
   (interactive "P")
-  (if (fboundp 'embark-dwim)
+  (if (fboundp bibtex-actions-at-point-function)
       (condition-case err
-          (if bibtex-actions-embark-dwim
-              (embark-dwim)
-            (embark-act))
+          (funcall bibtex-actions-at-point-function)
         (user-error
          (when (and (string-equal (error-message-string err) "No target found")
-                    bibtex-actions-at-point-fallback)
-           (bibtex-actions-run-default-action
-            (bibtex-actions-read :rebuild-cache arg)))))
-    (if-let ((keys (bibtex-actions-citation-key-at-point)))
-        (funcall bibtex-actions-default-action keys)
-      (when bibtex-actions-at-point-fallback
-        (bibtex-actions-run-default-action
-         (bibtex-actions-read :rebuild-cache arg))))))
+                    bibtex-actions-at-point-fallback) ;; fallback when no target is found
+           (let ((bibtex-actions-at-point-function)) ;; disable bibtex-actions-at-point-function and retry
+             (bibtex-actions-at-point arg)))))
+    (bibtex-actions-run-default-action ;; prompt directly when bibtex-actions-at-point-function is nil or not bound
+     (bibtex-actions-read :rebuild-cache arg))))
```

Comment:
```suggestion
    (if-let ((keys (bibtex-actions-citation-key-at-point)))
        (funcall bibtex-actions-default-action keys))))
```

Reply:
We should just run the default action on the citation keys if `bibtex-actions-at-point-function` is not available.

I think `bibtex-actions-at-point-fallback` should also be removed because (1) it depends on the specific error raised by `embark` and (2) it is unnecessary in an `at-point` function. I suggest the following 

```lisp
(defun bibtex-actions-at-point ()
  "Run the default action on citation keys found at point."
  (interactive)
  (if (fboundp bibtex-actions-at-point-function)
      (funcall bibtex-actions-at-point-function)
    (if-let ((keys (bibtex-actions-citation-key-at-point)))
        (funcall bibtex-actions-default-action keys))))
```
and set the default value of `bibtex-actions-at-point-function` to nil.

Reply:
Oh I am happy with that suggestion of yours @ilupin. This is much better than what I had proposed (I had also thought about this).

The good thing is you can take this suggestion to the natural conclusion and remove `bibtex-actions-at-point` entirely without loss of functionality :grimacing:

~~~ elisp
(defun bibtex-dwim ()
  "Run the default action on citation keys found at point."
  (interactive)
    (if-let ((keys (bibtex-actions-citation-key-at-point)))
        (funcall bibtex-actions-default-action keys)))
~~~

Then there is no need for a `bibtex-actions-at-point-function` or a `bibtex-actions-at-point`. The user either calls `embark-dwim`, `embark-act` or `bibtex-dwim`. This is actually the best solution and what I had originally in mind when asking @bdarcus why there is even a `bibtex-actions-at-point` command wrapping `embark-act/dwim`.

Reply:
WDYT @ilupin ?

And then on #174, how would that impact this key (now broken) function?

``` elisp
(defun bibtex-actions-org-cite-follow (_datum _arg)
  "Follow processor for org-cite."
  (call-interactively 'bibtex-actions-at-point))
```
What would we be calling here?

Maybe we need the defcustom for this case?

To remind: this is the function that hooks up to `org-open-at-point`.

Reply:
> But maybe we need the defcustom here, in case a user wants `embark-act`?

There is `bibtex-actions-at-point-function`, which can be called directly in the `:follow` processor. Users can set it to `bibtex-actions-dwim`(the default) or `embark-act/dwim`, then we don't need to wrap `embark-act/dwim` in any `bibtex-actions` functions.

```lisp
(defun bibtex-actions-org-cite-follow (_datum _arg)
  "Follow processor for org-cite."
  (call-interactively bibtex-actions-at-point-function))

;; or keep the original name `bibtex-actions-at-point'
(defun bibtex-actions-dwim ()
  "Run the default action on citation keys found at point."
  (interactive)
    (if-let ((keys (bibtex-actions-citation-key-at-point)))
        (funcall bibtex-actions-default-action keys)))
```

Reply:
I'm realizing there a couple of things specific to `org-cite` here we should address.

1. this
2. also, I mention in the main post that I think we should have a target finder specific to org-cite citations, so we can bind a different map to it

I'm hoping to sort all this out today, but I will probably merge #174 first, with hard-coded `embark-dwim`, and then finish this, and probably make the follow processor configurable there.

That's because this is more complicated.